### DISCUSSION TITLE
ES8388 stereo downmix (register 29)

### DISCUSSION THREAD

**Discussion by pmannk** [sle118/squeezelite-esp32#203] [Ideas]
Hi,
Firstly, thanks for this great project. Absolutely love it!

Could you please consider exposing an option to control Register 29 on the ES8388?
The datasheet indicates register 29, bit 5 can control the stereo/mono behaviour of the DAC.
![image](https://user-images.githubusercontent.com/79073108/202611186-2696bb8f-5941-4ccb-8376-648b2a5e7cb7.png)
If I'm understanding the data sheet correctly this looks like a great way to downmix a mono output, in hardware.

[Linking this old issue for reference in case anyone searches for it.](https://github.com/sle118/squeezelite-esp32/issues/36)
Thankyou.

**Comment by sle118** [comment]
You should try to configure speakeron event of dac_controlset

https://github.com/sle118/squeezelite-esp32#daci2s

**Comment by pmannk** [comment]
Thanks. Have had a couple of unsuccessful attempts. Will keep trying to work it out.

**Comment by sle118** [comment]
For now, I'm concentrating on some stabilizing of release 4.3 and, given we're already short on flash space, is unlikely that we'll be able to implement new features anytime soon.

Hopefully the events should accommodate this. I'll convert this to a conversation instead of an issue

**Comment by pmannk** [comment]
I've tried quite a few combinations with dac_controlset, unfortunately without success.
Based on the datasheet, I need to set register 29 to 00100000 (32), so I used:
`{ "speakeron": [ {"reg":29,"val":32} ] }`
This resulted in the board outputting no sound at all.
I backed this out, reset the board and confirmed that it would once again output sound.

Next I tried changing register 23. In theory bit 7 should swap the left and right channels:
![image](https://user-images.githubusercontent.com/79073108/203898671-2e27fee8-cc02-42ea-a0e2-27897615806b.png)
`{ "speakeron": [ {"reg":23,"val":128} ] }`
Again, as soon as I made this change the board wouldn't output sound at all.


I also tried separately testing both registers with a value of 0 (which is the default for both registers according to the datasheet). 
`{ "speakeron": [ {"reg":29,"val":0} ] }`
`{ "speakeron": [ {"reg":23,"val":0} ] }`
Again, the board wouldn't output sound.

I also tried without quoting speakeron  `{ speakeron: [ {"reg":29,"val":0} ] }` to no avail.
I'm at a loss why it's behaving this way.
